### PR TITLE
#74: Implement loss function

### LINKED ISSUE #74: Research + Implement Loss Function
The way restaurants are recommended depends on multiple factors: user dietary preferences, preferred dishes, cuisine type, and location.

We need to figure out a way to render restaurants with the highest match to user preferences, without any loss due to of how variable these preferences can be. 

This task relates to User Story #22

### PR DESCRIPTION
**Pull request by thomastai1666**
Closes #74. This PR implements a basic ranking algorithm to determine if a restaurant is a good match for a user. In the data folder are mock restaurant results from the documenu API, which will be implemented at a later date. We test the generate_restuaraunts_map and loss_function to make sure that they return the expected result for the two users.

Also adds a TODO for GET /restauraunts which is currently blocked until we set up documenu and capture the required data from users.

### CODE REVIEW

File: back-end/utils/loss_function.js [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+const loss_function = (restauraunts, user_preferences) => {
+  let restauraunt_ranking = generate_restuaraunts_map(restauraunts);
+  restauraunts.data.forEach((restauraunt) => {
+    user_preferences.forEach((user) => {
+      // Match user price preference
+      if (user.price === restauraunt.price_range) {
+        restauraunt_ranking[restauraunt.restaurant_id]["points"] += 1;
+      }
+      // Match user cuisine preference
+      user.cuisines.forEach((cuisine) => {
```
Comment (Jen-Lopez) [inline back-end/utils/loss_function.js:47]: For this, there's only one cuisine to match -- the "winning" cuisine that's been voted on by everyone.

Comment (Jen-Lopez) [reply back-end/utils/loss_function.js:47]: so I don't think matching by individual cuisine preference is necessary. We should only fetch restaurants that have the one cuisine.

Comment (thomastai1666) [reply back-end/utils/loss_function.js:47]: Sure thing! I'll remove this from the code and update the tests

**Review by Jen-Lopez — APPROVED** [review]

**Review by yunko1803 — APPROVED** [review]
LGTM!

**Pull request merged**